### PR TITLE
Update docs/cuda_custom_call (P1)

### PR DESCRIPTION
**Pull request by phu0ngng**
~~This PR added `CmdBufferCompatible` traits to the `cuda_custom_call` example, enabling cudaGraph for the custom calls.~~ 
In addition, custom calls are now registered via `jax.extend.ffi.register_ffi_target()`, as recommended by the latest JAX guidelines.

### CODE REVIEW

File: docs/cuda_custom_call/foo.cu.cc [hunk 1]
```diff
@@ -135,4 +136,5 @@ XLA_FFI_DEFINE_HANDLER_SYMBOL(
         .Arg<ffi::Buffer<ffi::DataType::F32>>()    // b_plus_1
         .Ret<ffi::Buffer<ffi::DataType::F32>>()    // a_grad
         .Ret<ffi::Buffer<ffi::DataType::F32>>()    // b_grad
-        .Attr<size_t>("n"));
+        .Attr<size_t>("n"),
+        {xla::ffi::Traits::kCmdBufferCompatible});  // cudaGraph enabled
```
Comment (andportnoy) [inline docs/cuda_custom_call/foo.cu.cc:140]: µnit: comment whitespace misaligned :)

**Review by andportnoy — COMMENTED** [review]
Thank you for helping keep it fresh and informative. I defer to @dfm to review for adherence to current best practices.

**Comment by dfm** [comment]
Thank you! Looks like you need to delete the `from jax.lib import xla_client` line, but otherwise this looks good to me.

Since the support for `CmdBufferCompatible` is broken in the currently released jaxlib (https://github.com/openxla/xla/pull/18360 hasn't been integrated into a release), I expect this will break CI.

So, how would you feel about splitting this in 2? One with the `register_ffi_target` change and another with the cmdbuffer change?

**Review by dfm — APPROVED** [review]
Approving to run internal CI.

**Comment by phu0ngng** [comment]
Thanks for the fast reviews. I will split it into 2 PR-s.

**Comment by phu0ngng** [comment]
The `CmdBufferCompatible` change went to #24345.

**Review by dfm — APPROVED** [review]

**Pull request merged**